great_expectations/expectations/expectation.py

```diff
@@ -2573,6 +2573,49 @@ def validate_configuration(
             raise InvalidExpectationConfigurationError(str(e))
 
 
+@public_api
+class ColumnExpectation(ColumnAggregateExpectation, ABC):
+    """Base class for column aggregate Expectations.
+
+    These types of Expectation produce an aggregate metric for a column, such as the mean, standard deviation,
+    number of unique values, column type, etc.
+
+    WARNING: This class is deprecated in favor of ColumnAggregateExpectation as of v0.16.6. It will be removed in a future release.
+    There is no change in functionality between the two classes; just a name change for clarity.
+
+    --Documentation--
+        - https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_aggregate_expectations/
+
+    Args:
+     domain_keys (tuple): A tuple of the keys used to determine the domain of the
+         expectation.
+     success_keys (tuple): A tuple of the keys used to determine the success of
+         the expectation.
+     default_kwarg_values (optional[dict]): Optional. A dictionary that will be used to fill unspecified
+         kwargs from the Expectation Configuration.
+
+         - A  "column" key is required for column expectations.
+
+    Raises:
+        InvalidExpectationConfigurationError: If no `column` is specified
+    """
+
+    def __init__(
+        self,
+        configuration: Optional[ExpectationConfiguration] = None,
+    ):
+        # deprecated-v0.16.6
+        warnings.warn(
```

Comment:
If we introduce a deprecation warning here now, it will raise warnings for any suite containing a custom (non-contributed) expectation. I think we should remove this warning for now and add it back in after we've had a chance to communicate the change (ie: after the conceptual guide is out)